lib/site-inspector.rb

```diff
@@ -127,37 +134,213 @@ def to_json
     to_hash.to_json
   end
 
-  def to_hash
+  def http
+    details = {
+      # site-inspector's best guess
+      canonical: uri.to_s,
+      canonical_protocol: https? ? :https : :http,
+      canonical_endpoint: www? ? :www : :root,
+
+      endpoints: endpoints
+    }
+
+    combos = details[:endpoints]
+
+    details[:live] = (
+      combos[:https][:www][:live] or
+      combos[:https][:root][:live] or
+      combos[:http][:www][:live] or
+      combos[:http][:root][:live]
+    )
+
+    details[:broken_root] = (
+      (combos[:https][:root][:status] == 0) and
+      (combos[:http][:root][:status] == 0)
+    )
+
+    details[:broken_www] = (
+      (combos[:https][:www][:status] == 0) and
+      (combos[:http][:www][:status] == 0)
+    )
+
+    details[:enforce_https] = (
+      (
+        (combos[:http][:www][:status] == 0) ||
+        (combos[:http][:www][:redirect_to_https])
+      ) and
+      (
+        (combos[:http][:root][:status] == 0) or
+        (combos[:http][:root][:redirect_to_https])
+      ) and
+      (
+        combos[:https][:www][:live] or
+        combos[:https][:root][:live]
+      )
+    )
+
+    details[:redirect] = (
+      combos[:http][:www][:redirect_to_external] and
+      combos[:http][:root][:redirect_to_external] and
+      combos[:https][:www][:redirect_to_external] and
+      combos[:https][:root][:redirect_to_external]
+    )
```

Comment:
A site is considered a "redirect domain" if all 4 of its combinations redirect to an external site.

This seems wrong, re-reading it -- a site could be a redirector even if only listens over HTTP. This should be tweaked so that the HTTPS endpoints have the option of being broken.